crates/dekaf/src/task_manager.rs

```diff
@@ -427,14 +428,25 @@ async fn update_partition_info(
             .as_ref()
             .context("expected partition template")?;
 
+        let template_name = partition_template.name.clone();
+        let task_name_clone = task_name.to_string();
+
         let partition_selector = binding
             .partition_selector
             .as_ref()
+            .map(|selector| {
+                let include = labels::add_value(
```

Comment:
You'll want to use set_value instead of add_value.

[The gazette optimization only kicks in if there's a single label value](https://github.com/gazette/core/blob/2580071332a6bf7f9302af1e513391f8c6539f5d/broker/list_apply_api.go#L147), so set_value makes this resilient to the partition selector including `name:prefix` as well.

Also, you're discarding `selector.exclude`, which potentially isn't empty.

Reply:
Thanks -- swapped to `set_value` and fixed droping `selector.exclude`